data/mods/gen6/random-teams.ts

```diff
@@ -1,520 +1,520 @@
+	['leechseed', 'substitute'],
+	['perishsong', 'protect'],
+	['solarbeam', 'sunnyday'],
+];
+
+/** Pokemon who always want priority STAB, and are fine with it as its only STAB move of that type */
+const PRIORITY_POKEMON = [
+	'aegislashblade', 'banette', 'breloom', 'cacturne', 'doublade', 'dusknoir', 'honchkrow', 'scizor', 'scizormega', 'shedinja',
 ];
 
 export class RandomGen6Teams extends RandomGen7Teams {
-	randomData: {[species: string]: OldRandomBattleSpecies} = require('./random-data.json');
+	randomSets: {[species: string]: RandomTeamsTypes.RandomSpeciesData} = require('./random-sets.json');
 
 	constructor(format: Format | string, prng: PRNG | PRNGSeed | null) {
 		super(format, prng);
+
 		this.noStab = NO_STAB;
 
 		this.moveEnforcementCheckers = {
 			Bug: (movePool, moves, abilities, types, counter) => (
 				['megahorn', 'pinmissile'].some(m => movePool.includes(m)) ||
-				!counter.get('Bug') && (abilities.has('Tinted Lens') || abilities.has('Adaptability'))
-			),
-			Dark: (movePool, moves, abilities, types, counter, species) => (
-				(!counter.get('Dark') && !abilities.has('Protean'))
-			),
-			Dragon: (movePool, moves, abilities, types, counter) => (
-				!counter.get('Dragon') &&
-				!abilities.has('Aerilate') &&
-				!abilities.has('Pixilate') &&
-				!moves.has('dragonascent') &&
-				!moves.has('rest') &&
-				!moves.has('sleeptalk')
+				!counter.get('Bug') && abilities.has('Tinted Lens')
 			),
+			Dark: (movePool, moves, abilities, types, counter) => !counter.get('Dark'),
+			Dragon: (movePool, moves, abilities, types, counter) => !counter.get('Dragon'),
 			Electric: (movePool, moves, abilities, types, counter) => !counter.get('Electric'),
-			Fairy: (movePool, moves, abilities, types, counter) => (
-				!counter.get('Fairy') && !abilities.has('Pixilate') && (!!counter.setupType || !counter.get('Status'))
-			),
-			Fighting: (movePool, moves, abilities, types, counter, species) => (
-				!counter.get('Fighting') && (
-					species.baseStats.atk >= 110 ||
-					abilities.has('Justified') || abilities.has('Unburden') ||
-					!!counter.setupType || !counter.get('Status')
-				)
-			),
-			Fire: (movePool, moves, abilities, types, counter) => !counter.get('Fire') ||
-				['eruption', 'quiverdance'].some(m => movePool.includes(m)),
-			Flying: (movePool, moves, abilities, types, counter) => (
-				!counter.get('Flying') && (
-					abilities.has('Gale Wings') ||
-					abilities.has('Serene Grace') ||
-					(types.has('Normal') && movePool.includes('bravebird'))
-				)
+			Fairy: (movePool, moves, abilities, types, counter) => !counter.get('Fairy'),
+			Fighting: (movePool, moves, abilities, types, counter) => !counter.get('Fighting'),
+			Fire: (movePool, moves, abilities, types, counter) => !counter.get('Fire'),
+			Flying: (movePool, moves, abilities, types, counter, species) => (
+				!counter.get('Flying') && ['aerodactylmega', 'charizardmegay', 'mantine', 'murkrow'].every(m => species.id !== m) &&
```

Comment:
```suggestion
				!counter.get('Flying') && !['aerodactylmega', 'charizardmegay', 'mantine', 'murkrow'].includes(species.id) &&
```

Reply:
(this was done but i can't mark as resolved)